core/tests/test_utils.py

```diff
@@ -622,6 +622,9 @@ def process_and_flush_pending_tasks(self, queue_name=None):
                 if task.url == '/_ah/queue/deferred':
                     from google.appengine.ext import deferred
                     deferred.run(task.payload)
+                elif feconf.EMAILS_TASK_PREFIX in task.url:
+                    # Flush task if task is of emails taskqueue.
```

Comment:
Well gae forces that a taskqueue payload must be an instance of string, which is not in this case.
We need user_id because without it, how can we get the email data model, which contains information about everything.